plugin/src/Modifier.cc

```diff
@@ -927,6 +927,172 @@ Mod_As_Duration::load(Config &cfg, YAML::Node, TextView, TextView, YAML::Node ke
   return Handle(new self_type{std::move(expr)});
 }
 
+// ---
+/// url-encode modifier.
+class Mod_url_encode : public Modifier
+{
+  using self_type  = Mod_url_encode;
+  using super_type = Modifier;
+
+  // Our own map. TSStringPercentEncode will not escape all we need.
+  static constexpr unsigned char escape_codes[32] = {
+    0xFF, 0xFF, 0xFF,
+    0xFF,       // control
+    0xBE,       // space ” # % $ &
+    0x19,       // + , /
+    0x00,       //
+    0x3F,       // < > : ; = ?
+    0x80,       // @
+    0x00, 0x00, //
+    0x1E, 0x80, // [ \ ] ^ `
+    0x00, 0x00, //
+    0x1F,       // { | } ~ DEL
+    0x00, 0x00, 0x00,
+    0x00, // all non-ascii characters unmodified
+    0x00, 0x00, 0x00,
+    0x00, //
+    0x00, 0x00, 0x00,
+    0x00, //
+    0x00, 0x00, 0x00,
+    0x00 //
+  };
+
+public:
+  inline static const std::string KEY = "url-encode";
+  /** Check if @a ftype is a valid type to be modified.
+   *
+   * @param ftype Type of feature to modify.
+   * @return @c true if this modifier can modity that feature type, @c false if not.
+   */
+  bool is_valid_for(ActiveType const &ex_type) const override;
+
+  /// Resulting type of feature after modifying.
+  ActiveType result_type(ActiveType const &) const override;
+
+  /** Modify the feature.
+   *
+   * @param ctx Run time context.
+   * @param feature Feature to modify [in,out]
```

Comment:
I think that's archaic - IIRC originally the `Feature` was passed as a reference, but at some point I decided that was too dangerous and switch to returning it. So it's just an "in" parameter.